Feature/validate nameing conventions

SebastianSchildt opened the pull request:
Codifies best-practices: Upon parsing vpsec check whether VSS naming conventions from https://covesa.github.io/vehicle_signal_specification/rule_set/basics/ are followed.

Will only print warnings, as not following this recommendations is still valid VSS.

This also changes the main parser to keep the hints for the origin .vspec file in the model (they have always been added, but before had been removed in a last step). This enables printing more helpful messages, and does not influence generated outputs


Run with current model gives

```
WARNING: Node "bodyType" found in file "Vehicle/Vehicle.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Node "dateVehicleFirstRegistered" found in file "Vehicle/Vehicle.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Node "meetsEmissionStandard" found in file "Vehicle/Vehicle.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Node "productionDate" found in file "Vehicle/Vehicle.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Node "purchaseDate" found in file "Vehicle/Vehicle.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Node "vehicleModelDate" found in file "Vehicle/Vehicle.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Node "vehicleConfiguration" found in file "Vehicle/Vehicle.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Node "vehicleSeatingCapacity" found in file "Vehicle/Vehicle.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Node "vehicleSpecialUsage" found in file "Vehicle/Vehicle.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Node "vehicleinteriorColor" found in file "Vehicle/Vehicle.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Node "vehicleinteriorType" found in file "Vehicle/Vehicle.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Node "knownVehicleDamages" found in file "Vehicle/Vehicle.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Node "cargoVolume" found in file "Vehicle/Vehicle.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Node "emissionsCO2" found in file "Vehicle/Vehicle.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Boolean node "Connected" found in file "Vehicle/Vehicle.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "EngineStopStartEnabled" found in file "Powertrain/FuelSystem.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "LowFuelLevel" found in file "Powertrain/FuelSystem.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Node "intensity" found in file "Body/Body.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Boolean node "Status" found in file "Body/Body.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "LevelLow" found in file "Body/Body.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Status" found in file "Body/Body.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "LevelLow" found in file "Body/Body.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Status" found in file "ExteriorMirrors.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Status" found in file "ExteriorMirrors.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "isOpen" found in file "SingleWindow.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Node "isOpen" found in file "SingleWindow.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Boolean node "ChildLock" found in file "SingleWindow.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "isOpen" found in file "SingleWindow.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Node "isOpen" found in file "SingleWindow.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Boolean node "ChildLock" found in file "SingleWindow.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "isOpen" found in file "SingleWindow.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Node "isOpen" found in file "SingleWindow.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Boolean node "ChildLock" found in file "SingleWindow.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "isOpen" found in file "SingleWindow.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Node "isOpen" found in file "SingleWindow.vspec" is not following naming conventions. It is recommended that node names use camel case, starting with a capital letter, only using letters A-z and numbers 0-9.
WARNING: Boolean node "ChildLock" found in file "SingleWindow.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "HasPassenger" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Warmer" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Cooler" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Increase" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Decrease" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Inflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Deflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Inflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Deflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "HasPassenger" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Warmer" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Cooler" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Increase" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Decrease" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Inflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Deflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Inflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Deflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "HasPassenger" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Warmer" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Cooler" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Increase" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Decrease" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Inflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Deflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Inflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Deflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "HasPassenger" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Warmer" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Cooler" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Increase" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Decrease" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Inflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Deflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Inflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Deflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "HasPassenger" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Warmer" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Cooler" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Increase" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Decrease" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Inflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Deflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Inflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Deflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "HasPassenger" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Warmer" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Cooler" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Increase" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Decrease" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Forward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Backward" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Up" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Down" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Inflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Deflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Inflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Deflate" found in file "SingleSeat.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Error" found in file "ADAS/ADAS.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Warning" found in file "ADAS/ADAS.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Error" found in file "ADAS/ADAS.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Error" found in file "ADAS/ADAS.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Error" found in file "ADAS/ADAS.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Error" found in file "ADAS/ADAS.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "Error" found in file "ADAS/ADAS.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "FluidLevelLow" found in file "Wheel.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "BrakesWorn" found in file "Wheel.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "PressureLow" found in file "Wheel.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "FluidLevelLow" found in file "Wheel.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "BrakesWorn" found in file "Wheel.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "PressureLow" found in file "Wheel.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "FluidLevelLow" found in file "Wheel.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "BrakesWorn" found in file "Wheel.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "PressureLow" found in file "Wheel.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "FluidLevelLow" found in file "Wheel.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "BrakesWorn" found in file "Wheel.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "PressureLow" found in file "Wheel.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "MIL" found in file "OBD/OBD.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "AuxInputStatus" found in file "OBD/OBD.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "MIL" found in file "OBD/OBD.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "EyesOnRoad" found in file "Driver/Driver.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
WARNING: Boolean node "ServiceDue" found in file "Vehicle/Service.vspec" is not following naming conventions. It is recommended that boolean nodes start with "Is".
```

erikbosch commented on vspec/__init__.py:
```diff
@@ -308,9 +308,7 @@ def cleanup_flat_entries(flat_model):
 def cleanup_deep_model(deep_model):
     # Traverse the flat list of the parsed specification
     # Is this an include element?
```
Does the comment:

`# Is this an include element?`

.. belong to the code you removed? Or also to remaining lines?

SebastianSchildt replied:
I wasn't sure what the comment meant, so I kept it. I did not see any relation to includes int the whole function

The comment is from @magnusfeuer  6years, the code below was last touched by @danielwilms. Maybe one of them knows.

SebastianSchildt replied:
Removed, I think this comment was an artefact

oliveirarleo commented on vspec/model/vsstree.py:
```diff
@@ -73,6 +73,9 @@ def __init__(self, name, source_dict: dict, parent=None, children=None, break_on
         self.type = VSSType.from_str(source_dict["type"])
         self.uuid = source_dict["uuid"]
 
+        if "$file_name$" not in source_dict.keys():
+            source_dict["$file_name$"]="UNKNOWN"
+
```
In my opinion, `$file_name$` should use `source_dict["$file_name$"]=None` and then either guarantee to have `$file_name$` or use something along these lines:
```python
if "$file_name$" in source_dict:
    raise FileNameNotFoundException
```

Or else if this continues like this, later it could show errors like  `Warning: UNKNOWN filename does not follow the vss naming pattern`

oliveirarleo replied:
In fact i believe to use $file_name$ in the source_dict is a bad idea in general. It's better to just send as a mandatory argument in the constructor.

SebastianSchildt replied:
let it crash->That may be useful (instead of "hiding" some error).

 Removing it from source_dict: I do not think so, _if_ I understand the code correctly the source dict is the (preprocessed) YAML, that is is extended with additional metadata during earlier parsing steps, so it really helps traceability to have such information "directly" near the data, should problem arise, and not splitting it out and "manually" taking care of this association.

SebastianSchildt replied:
Removed the check. Now it will crash and burn, if $file_name$ is not there instead of hiding the error.

SebastianSchildt commented on vspec/model/vsstree.py:
```diff
@@ -103,6 +103,22 @@ def __init__(self, name, source_dict: dict, parent=None, children=None, break_on
 
+        self.check_style(source_dict)
+
+    def check_style(self,source_dict):
```
change  method name and string as parameter sounds good. Not sure about "static" method, as the check needs to know the type of the current node. (I think not useful to put that all in parameters)

oliveirarleo replied:
That's true, my bad, so `validate_style(self, name: str)` should be best

SebastianSchildt commented on vspec/model/vsstree.py:
```diff
@@ -103,6 +103,22 @@ def __init__(self, name, source_dict: dict, parent=None, children=None, break_on
         if "comment" in source_dict.keys():
             self.comment = source_dict["comment"]
 
+        self.check_style(source_dict)
```
Actually I was thinking about that as well, but was too lazy :) 
 If people agree it is useful I might add it as another keyword parameter in the VSSTree parse code. But I would probably not adapt all tools.

oliveirarleo replied:
I understand hahaha

Well, in that case

```python
try:
    self.validate_style(source_dict["$file_name$"])
except FileNameStyleException as e:
        print(f"Warning: {e}")
```
is a good alternative

SebastianSchildt replied:
Done

SebastianSchildt commented:
Now we have a flag for checking name style, implemented the same way as the breaking flag for non-core attributes. Does not need to be set, in that case defaults to only print a warning.

Modified vspec2json to allow individual setting of flags, such as

```
usage: vspec2json.py [-h] [-I dir] [-s] [--abort-on-non-core-attribute] [--abort-on-name-style] [--no-uuid] <vspec_file> <json file>

Convert vspec to json.

positional arguments:
  <vspec_file>          The vehicle specification file to convert.
  <json file>           The file to output the JSON objects to.

optional arguments:
  -h, --help            show this help message and exit
  -I dir, --include-dir dir
                        Add include directory to search for included vspec files.
  -s, --strict          Use strict checking: Terminate when anything not covered or not recommended by the core VSS specs is found.
  --abort-on-non-core-attribute
                        Terminate when non-core attribute is found.
  --abort-on-name-style
                        Terminate naming style not follows recommendations.
  --no-uuid             Exclude uuids from generated json.
```

the "--strict" flag is equivalent to setting both abort flags.

Extended the strict flag in 2csv and 2yaml to also include both aborts. Did not add the individual flags (too much copy&paste, and should be all rolled into one codebase anyway)

Did not change 2franca and 2binary, as they are still using "old-style" parsing, and thus are not compatible with those flags anyway

danielwilms commented:
@SebastianSchildt fixes https://github.com/COVESA/vss-tools/pull/128#discussion_r778152339

Can be merged afterwards.

SebastianSchildt commented:
Fixed now @erikbosch  @danielwilms
